### PR TITLE
Display 'readonly' for record structs

### PR DESCRIPTION
**Pull request by Youssef1313**
Part of #53631.

### CODE REVIEW

**Review by RikkiGibson — APPROVED** [review]

**Comment by RikkiGibson** [comment]
@jcouv for compiler sign-off

@dotnet/roslyn-ide for a small test sign-off

File: src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs [hunk 1]
```diff
@@ -7255,6 +7255,19 @@ void M($$Person p)
 }", MainDescription("record struct Person"));
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task QuickInfoReadOnlyRecordStruct()
+        {
+            await TestWithOptionsAsync(
+                Options.Regular.WithLanguageVersion(LanguageVersion.CSharp9),
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs:7262]: I just noticed I unintentionally left the language version from copy/pasting the test above. But the test should probably fail with C# 9.0

However, It's passing. Feels weird.

File: src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs [hunk 2]
```diff
@@ -670,6 +669,15 @@ private void AddTypeKind(INamedTypeSymbol symbol)
                             break;
 
                         case TypeKind.Struct when symbol.IsRecord:
+                            // In case ref record structs are allowed in future, call AddKeyword(SyntaxKind.RefKeyword) and remove assertion.
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs:672]: The comment probably isn't needed

File: src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs (hunk 1 above)
Comment (jcouv) [reply src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs:7262]: That's expected. It should pass in C# 9. With some rare exceptions, we always bind in the latest version of the language, only producing different diagnostics. So the symbol is the same regardless of language version.

File: src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs (hunk 2 above)
Comment (Youssef1313) [inline src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs:672]:
```suggestion
```

**Comment by jcouv** [comment]
Merged/squashed. Thanks @Youssef1313

**Pull request merged**